version: "3.9"

services:
  db:
    image: postgres
    container_name: slavello_db
    restart: always
    environment:
      POSTGRES_DB: "slavello_db"
      POSTGRES_USER: "moderator"
      POSTGRES_PASSWORD: "2222"
    volumes:
      - ./scripts/migrations/schema.sql:/docker-entrypoint-initdb.d/1.sql
      - ./scripts/migrations/init_data.sql:/docker-entrypoint-initdb.d/2.sql
    ports:
      - "5432:5432"
    networks:
      - slavello-network
    deploy:
      restart_policy:
        condition: on-failure

  sessions-db:
    image: redis:alpine3.18
    container_name: slavello_sessions
    restart: always
    environment:
      REDIS_PASSWORD: "1234"
    command: redis-server --requirepass "${REDIS_PASSWORD}"
    ports:
      - "6379:6379"
    networks:
      - slavello-network
    deploy:
      restart_policy:
        condition: on-failure

  api:
    image: slavello_api
    build:
      context: .
      dockerfile: ./cmd/api/Dockerfile
    container_name: slavello_api
    ports:
      - "8000:8000"
    depends_on:
      - db
      - sessions-db
    volumes:
      - ./configs/api_main.yaml:/configs/api.yaml
    networks:
      - slavello-network
    deploy:
      restart_policy:
        condition: on-failure

  dev-frontend:
    image: node:19
    container_name: slavello_dev_frontend
    working_dir: /code/
    command: npm run start
    volumes:
      - ./frontend:/code/
    ports:
      - "8100:8100"
    networks:
      - slavello-network
    deploy:
      restart_policy:
        condition: on-failure

networks:
  slavello-network:
    name: slavello_network
    driver: bridge
